engine/servermaster/server.go

```diff
@@ -249,44 +247,64 @@ func (s *Server) Heartbeat(ctx context.Context, req *pb.HeartbeatRequest) (*pb.H
-	resp2 := &pb.SubmitJobResponse{}
-	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &resp2)
+// CreateJob delegates request to leader's JobManager.CreateJob.
+func (s *Server) CreateJob(ctx context.Context, req *pb.CreateJobRequest) (*pb.Job, error) {
+	job := &pb.Job{}
+	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &job)
 	if shouldRet {
-		return resp2, err
+		return job, err
 	}
-	return s.jobManager.SubmitJob(ctx, req), nil
+	return s.jobManager.CreateJob(ctx, req)
 }
 
-// QueryJob implements pb.MasterServer.QueryJob
-func (s *Server) QueryJob(ctx context.Context, req *pb.QueryJobRequest) (*pb.QueryJobResponse, error) {
-	resp2 := &pb.QueryJobResponse{}
-	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &resp2)
+// GetJob delegates request to leader's JobManager.GetJob.
+func (s *Server) GetJob(ctx context.Context, req *pb.GetJobRequest) (*pb.Job, error) {
+	job := &pb.Job{}
+	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &job)
 	if shouldRet {
-		return resp2, err
+		return job, err
 	}
-	return s.jobManager.QueryJob(ctx, req), nil
+	return s.jobManager.GetJob(ctx, req)
 }
 
-// CancelJob implements pb.MasterServer.CancelJob
-func (s *Server) CancelJob(ctx context.Context, req *pb.CancelJobRequest) (*pb.CancelJobResponse, error) {
-	resp2 := &pb.CancelJobResponse{}
-	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &resp2)
+// ListJobs delegates request to leader's JobManager.ListJobs.
+func (s *Server) ListJobs(ctx context.Context, req *pb.ListJobsRequest) (*pb.ListJobsResponse, error) {
+	resp := &pb.ListJobsResponse{}
+	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &resp)
 	if shouldRet {
-		return resp2, err
+		return resp, err
 	}
-	return s.jobManager.CancelJob(ctx, req), nil
+	return s.jobManager.ListJobs(ctx, req)
 }
 
-// PauseJob implements pb.MasterServer.PauseJob
-func (s *Server) PauseJob(ctx context.Context, req *pb.PauseJobRequest) (*pb.PauseJobResponse, error) {
-	resp2 := &pb.PauseJobResponse{}
-	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &resp2)
+// PauseJob delegates request to leader's JobManager.PauseJob.
+func (s *Server) PauseJob(ctx context.Context, req *pb.PauseJobRequest) (*pb.Job, error) {
+	job := &pb.Job{}
+	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &job)
 	if shouldRet {
-		return resp2, err
+		return job, err
+	}
+	return s.jobManager.PauseJob(ctx, req)
+}
+
+// CancelJob delegates request to leader's JobManager.CancelJob.
+func (s *Server) CancelJob(ctx context.Context, req *pb.CancelJobRequest) (*pb.Job, error) {
+	job := &pb.Job{}
+	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &job)
+	if shouldRet {
+		return job, err
+	}
+	return s.jobManager.CancelJob(ctx, req)
+}
+
+// DeleteJob delegates request to leader's JobManager.DeleteJob.
+func (s *Server) DeleteJob(ctx context.Context, req *pb.DeleteJobRequest) (*emptypb.Empty, error) {
+	empty := &emptypb.Empty{}
+	shouldRet, err := s.masterRPCHook.PreRPC(ctx, req, &empty)
+	if shouldRet {
+		return empty, err
 	}
```

Comment:
`shouldRet` and `err` logic are confused. It's not introduced by this PR, we shall refactor it later. No need to address.